content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1525 @@
+so this will always get through provided the request is properly rewritten), 
+which will then redirect the user to another page in Jenkins (this only works
+correctly if you configured the response rewriting correctly), which then returns 200.
+
+This error message indicates that this test is failing - and the most
+likely cause is that the response rewriting is misconfigured. 
+See the *Server Configuration Guides* (below) for additional tips about
+configuring a reverse proxy. 
+
+NOTE: The reverse proxy tests were improved in release 1.552 so users
+with previously working proxy setups may start to receive proxy
+warnings. 
+
+Be sure to set the `+X-Forwarded-Proto+` header if your reverse proxy is
+accessed via HTTPS and then Jenkins itself is accessed via HTTP i.e.
+proxying HTTPS to HTTP.
+
+Changing the context path of Jenkins with a reverse proxy is fraught with danger. 
+There are lots of URLs that you need to rewrite correctly,
+and even if you get the ones in HTML files you may miss some in
+javascript, CSS or XML resources.
+
+The recommendation is to ensure that Jenkins is running at the context
+path that your reverse proxy is serving Jenkins at. 
+You will have the least pain if you keep to this principle.
+
+While it is technically possible to use rewrite rules to change the context path, 
+you should be aware that it would be a lot of work to find and fix everything in 
+your rewrite rules and the reverse proxy will spend most of its time rewriting
+responses from Jenkins. 
+Much easier to change Jenkins to run at the context path your reverse proxy is
+expecting, e.g. if your reverse proxy is forwarding requests at
+https://manchu.example.org/foobar/ to Jenkins then you could just use
+`+java -jar jenkins.war --prefix /foobar+` to start jenkins using
+`+/foobar+` as the context path
+
+[[Jenkinssaysmyreverseproxysetupisbroken-FurtherDiagnosis]]
+==== Further Diagnosis
+
+For further diagnosis, try using cURL:
+
+[source,sh]
+----
+curl -iL -e http://your.reverse.proxy/jenkins/manage \
+   http://your.reverse.proxy/jenkins/administrativeMonitor/hudson.diagnosis.ReverseProxySetupMonitor/test
+----
+
+(assuming your Jenkins is located at
+http://your.reverse.proxy/jenkins/ - and is open to anonymous read
+access)
+
+[[Jenkinssaysmyreverseproxysetupisbroken-ServerConfigurationGuides]]
+==== Server Configuration Guides
+
+While the pages talk primarily about Apache / NGinX / HAProxy / Squid,
+they also have information that applies to other reverse proxies.
+
+* <<Running Jenkins behind Apache>>
+* <<Running Jenkins behind Nginx>>
+* <<Running Jenkins behind HAProxy>>
+* <<Running Jenkins behind Squid>>
+* <<Running Jenkins behind IIS>>
+
+If using Apache check that _nocanon_ is set on _ProxyPass_ and
+that _AllowEncodedSlashes_ is set as per the Apache link above.
+
+_AllowEncodedSlashes_ is not inherited in Apache configs, so this
+directive must be placed inside the _VirtualHost_ definition.
```

Comment:
Could you move this elements in the "Running Jenkins behind Apache" section please?